src/main/java/com/questhelper/quests/treegnomevillage/TreeGnomeVillage.java

```diff
@@ -0,0 +1,207 @@
+
+		ConditionalStep defeatTheWarlord = new ConditionalStep(this, talkToTheWarlord,
+			new ItemRequirement("Food", -1),
+			new ItemRequirement("Armor & Weapons", -1));
+		defeatTheWarlord.addStep(fightingWarlord, fightTheWarlord);
+
+		return defeatTheWarlord;
+	}
+
+	private void setupItemRequirements()
+	{
+		logRequirement = new ItemRequirement("Logs", ItemID.LOGS, 6);
+	}
+
+	private void setupZones()
+	{
+		upstairsTower = new Zone(new WorldPoint(2500, 3251, 1), new WorldPoint(2506, 3259, 1));
+	}
+
+	public void setupConditions()
+	{
+		notCompleteFirstTracker = new VarbitCondition(TRACKER_1_VARBITID, 0);
+		notCompleteSecondTracker = new VarbitCondition(TRACKER_2_VARBITID, 0);
+		notCompleteThirdTracker = new VarbitCondition(TRACKER_3_VARBITID, 0);
+
+		completeFirstTracker = new VarbitCondition(TRACKER_1_VARBITID, 1);
+		completeSecondTracker = new VarbitCondition(TRACKER_2_VARBITID, 1);
+		completeThirdTracker = new VarbitCondition(TRACKER_3_VARBITID, 1);
+
+		isUpstairsTower = new ZoneCondition(upstairsTower);
+
+		talkToFirstTracker = new Conditions(notCompleteFirstTracker);
+		talkToSecondTracker = new Conditions(LogicType.AND, completeFirstTracker, notCompleteSecondTracker);
+		talkToThirdTracker = new Conditions(LogicType.AND, completeFirstTracker, notCompleteThirdTracker);
+		shouldFireBallista = new Conditions(LogicType.AND, completeFirstTracker, completeSecondTracker, completeThirdTracker);
+	}
+
+	private void setupSteps()
+	{
+		talkToKingBolren = new NpcStep(this, NpcID.KING_BOLREN, new WorldPoint(2541, 3170, 0), "Speak to King Bolren in the centre of the Tree Gnome Maze.");
+		talkToKingBolren.addDialogStep("Can I help at all?");
+		talkToKingBolren.addDialogStep("I would be glad to help.");
+
+		talkToCommanderMontai = new NpcStep(this, NpcID.COMMANDER_MONTAI, new WorldPoint(2523, 3208, 0), "Speak with Commander Montai.");
+		talkToCommanderMontai.addDialogStep("Ok, I'll gather some wood.");
+
+		bringWoodToCommanderMontai = new NpcStep(this, NpcID.COMMANDER_MONTAI, new WorldPoint(2523, 3208, 0), "Speak with Commander Montai again to give him the wood.");
+
+		talkToCommanderMontaiAgain = new NpcStep(this, NpcID.COMMANDER_MONTAI, new WorldPoint(2523, 3208, 0), "Speak with Commander Montai.");
+		talkToCommanderMontaiAgain.addDialogStep("I'll try my best.");
+
+		firstTracker = new NpcStep(this, NpcID.TRACKER_GNOME_1, new WorldPoint(2501, 3261, 0), "Talk to the first tracker gnome to the northwest.");
+		secondTracker = new NpcStep(this, NpcID.TRACKER_GNOME_2, new WorldPoint(2524, 3257, 0), "Talk to the second tracker gnome inside the jail.");
+		thirdTracker = new NpcStep(this, NpcID.TRACKER_GNOME_3, new WorldPoint(2497, 3234, 0), "Talk to the third tracker gnome to the southwest.");
+
+		fireBallista = new ObjectStep(this, ObjectID.BALLISTA, new WorldPoint(2509, 3211, 0), "Fire the ballista.");
```

Comment:
I believe varbit 602 relates to the ballista height, with 0 being 0001, 1 0002, 2 0003, and 3 0004. Would be good to have a conditional based on this to highlight the right option.

Reply:
I didn't detect any varbit changes regarding this.

Reply:
I'd guess that's due to you having 0, so it stayed the same. If you had a different answer to 0001 let me know.

Reply:
I had 2 as the degree.

Reply:
Just tested on another account, 602 went 0->1, then the correct co-ordinate was 0002, so definitely looks like it does represent it.

Reply:
Added, with a fallback step incase something goes weird.